### PR TITLE
Remove dependency on github.com/docker/docker

### PR DESCRIPTION
**Pull request by andrewkroh**
Docker is used to test in environments with different hostnames and domain names. This was done by interacting with the Docker API via a dependency on github.com/docker/docker. But this brings in a lot of dependency baggage.

This change replaces docker API interaction with invocation of `docker run`. This results in a reduced dependency set for the module.

_I recommend to hide whitespace changes when viewing the diff._

### CODE REVIEW

File: providers/linux/host_fqdn_integration_linux_test.go [hunk 1]
```diff
@@ -39,145 +37,96 @@ const (
...
+// if docker run returns a non-zero exit code.
+func dockerRun(t *testing.T, hostname, domain string, command []string) {
+	t.Helper()
 
-	pwd, err := os.Getwd()
-	if err != nil {
-		t.Fatalf("could not get current directory: %v", err)
+	if _, err := execabs.LookPath("docker"); err != nil {
+		t.Skipf("Skipping because docker was not found: %v", err)
 	}
```
Comment (efd6) [inline providers/linux/host_fqdn_integration_linux_test.go:96]: Hoist this to the outer loop? If it's not here once, it's likely not here at all.

File: providers/linux/host_fqdn_integration_linux_test.go [hunk 2]
```diff
@@ -39,145 +37,96 @@ const (
...
-		t.Fatalf("could not get container logs: %v", err)
+	if domain != "" {
+		args = append(args, "--domainname="+domain)
+	}
+	args = append(args, "golang:"+goVersion)
+	args = append(args, command...)
+
+	cmd := execabs.Command("docker", args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
```
Comment (efd6) [inline providers/linux/host_fqdn_integration_linux_test.go:126]: Capture these in a buffer? Then they can be emitted in the `t.Logf` call.

**Review by efd6 — COMMENTED** [review]
:tada: :scissors: :tada:

File: providers/linux/host_fqdn_integration_linux_test.go (hunk 2 above)
Comment (andrewkroh) [reply providers/linux/host_fqdn_integration_linux_test.go]: SGTM. Here's the new output:

```
ubuntu@neat-chimpanzee:/go-sysinfo/providers/linux$ go test  -run TestHost_FQDN -tags integration -v .
=== RUN   TestHost_FQDN
=== RUN   TestHost_FQDN/TestHost_FQDN_set_hostname+domainname
    host_fqdn_integration_linux_test.go:84: Running docker container using ["run" "--rm" "-v" "/home/ubuntu/go:/go" "-v" "/go-sysinfo:/go-sysinfo" "-w=/go-sysinfo" "--hostname=hostname" "--domainname=some.domain" "golang:1.22.2" "go" "test" "-v" "-tags" "integration,docker" "-run" "^TestHost_FQDN_set$" "./providers/linux"]
    host_fqdn_integration_linux_test.go:84: Container output:
        === RUN   TestHost_FQDN_set
        --- PASS: TestHost_FQDN_set (0.00s)
        PASS
        ok      github.com/elastic/go-sysinfo/providers/linux   0.005s
    host_fqdn_integration_linux_test.go:84: Exiting container
=== RUN   TestHost_FQDN/TestHost_FQDN_set_hostname_only
    host_fqdn_integration_linux_test.go:84: Running docker container using ["run" "--rm" "-v" "/home/ubuntu/go:/go" "-v" "/go-sysinfo:/go-sysinfo" "-w=/go-sysinfo" "--hostname=hostname.some.domain" "golang:1.22.2" "go" "test" "-v" "-tags" "integration,docker" "-run" "^TestHost_FQDN_set$" "./providers/linux"]
    host_fqdn_integration_linux_test.go:84: Container output:
        === RUN   TestHost_FQDN_set
        --- PASS: TestHost_FQDN_set (0.00s)
        PASS
        ok      github.com/elastic/go-sysinfo/providers/linux   0.006s
    host_fqdn_integration_linux_test.go:84: Exiting container
=== RUN   TestHost_FQDN/TestHost_FQDN_not_set
    host_fqdn_integration_linux_test.go:84: Running docker container using ["run" "--rm" "-v" "/home/ubuntu/go:/go" "-v" "/go-sysinfo:/go-sysinfo" "-w=/go-sysinfo" "golang:1.22.2" "go" "test" "-v" "-count" "1" "-tags" "integration,docker" "-run" "^TestHost_FQDN_not_set$" "./providers/linux"]
    host_fqdn_integration_linux_test.go:84: Container output:
        === RUN   TestHost_FQDN_not_set
        --- PASS: TestHost_FQDN_not_set (0.00s)
        PASS
        ok      github.com/elastic/go-sysinfo/providers/linux   0.007s
    host_fqdn_integration_linux_test.go:84: Exiting container
--- PASS: TestHost_FQDN (54.56s)
    --- PASS: TestHost_FQDN/TestHost_FQDN_set_hostname+domainname (17.82s)
    --- PASS: TestHost_FQDN/TestHost_FQDN_set_hostname_only (18.14s)
    --- PASS: TestHost_FQDN/TestHost_FQDN_not_set (18.60s)
PASS
ok      github.com/elastic/go-sysinfo/providers/linux   54.562s
```

**Review by efd6 — APPROVED** [review]

**Pull request merged**